langchain/src/callbacks/handlers/tracers.ts

```diff
@@ -114,14 +111,43 @@ export abstract class BaseTracer extends BaseCallbackHandler {
     this.runMap.set(run.uuid, run);
   }
 
-  protected async _endTrace(run: LLMRun | ChainRun | ToolRun): Promise<void> {
+  protected async _endTrace(run: Run): Promise<void> {
     if (!run.parent_uuid) {
       await this.persistRun(run);
-      this.executionOrder = 1;
     }
     this.runMap.delete(run.uuid);
   }
 
+  protected _getExecutionOrder(parentRunId: string | undefined): number {
+    // If a run has no parent then execution order is 1
+    if (parentRunId === undefined) {
+      return 1;
+    }
+
+    const parentRun = this.runMap.get(parentRunId);
+
+    if (parentRun === undefined) {
+      throw new Error(`Parent run ${parentRunId} not found`);
+    }
+
+    // If a run has a parent then execution order is
+    // parent's execution order + (count of previous child runs) + 1
+
+    let executionOrder = parentRun.execution_order;
+
+    if (parentRun.type === "chain" || parentRun.type === "tool") {
```

Comment:
think we should raise an error if the parent run type is "llm", that shouldn't be possible.